support/agent/cases_test.go

```diff
@@ -177,7 +185,150 @@ func TestCommitPowerAndCheckInvariants(t *testing.T) {
 
 		epoch := sim.GetVM().GetEpoch()
 		if epoch%100 == 0 {
-			stateTree, err := sim.GetVM().GetStateTree()
+			stateTree, err := sim.GetVM().(*vm_test.VM).GetStateTree()
+			require.NoError(t, err)
+
+			totalBalance, err := sim.GetVM().(*vm_test.VM).GetTotalActorBalance()
+			require.NoError(t, err)
+
+			acc, err := states.CheckStateInvariants(stateTree, totalBalance, sim.GetVM().GetEpoch()-1)
+			require.NoError(t, err)
+			require.True(t, acc.IsEmpty(), strings.Join(acc.Messages(), "\n"))
+
+			require.NoError(t, sim.GetVM().GetState(builtin.StoragePowerActorAddr, &pwrSt))
+
+			// assume each sector is 32Gb
+			sectorCount := big.Div(pwrSt.TotalBytesCommitted, big.NewInt(32<<30))
+
+			fmt.Printf("Power at %d: raw: %v  cmtRaw: %v  cmtSecs: %d  cnsMnrs: %d avgWins: %.3f  msgs: %d\n",
+				epoch, pwrSt.TotalRawBytePower, pwrSt.TotalBytesCommitted, sectorCount.Uint64(),
+				pwrSt.MinerAboveMinPowerCount, float64(sim.WinCount)/float64(epoch), sim.MessageCount)
+		}
+	}
+}
+
+func TestMigration(t *testing.T) {
+	t.Skip("slow")
```

Comment:
Can we bind this to an environment variable and then test it in a separate CI job?

Reply:
The test flags from go-filecoin (venus) worked well. Maybe time to replicate that here.

Reply:
In general👍  
I want to get this overdue pr merged and do agent testing right so I've written up an issue to track this #1385